controllers/infinispan_service_controller.go

```diff
@@ -538,6 +554,39 @@ func computeSiteService(ispn *ispnv1.Infinispan) *corev1.Service {
 	return &siteService
 }
 
+func computeSiteRoute(ispn *ispnv1.Infinispan) *routev1.Route {
+	host := strings.TrimSpace(ispn.Spec.Service.Sites.Local.Expose.RouteHostName)
+	route := routev1.Route{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "route.openshift.io/v1",
+			Kind:       "Route",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ispn.GetSiteRouteName(),
+			Namespace: ispn.Namespace,
+			Labels:    LabelsResource(ispn.Name, "infinispan-route-xsite"),
+		},
+		Spec: routev1.RouteSpec{
+			Host: host,
```

Comment:
What if the `host` is empty?

Reply:
Right, I was more wondering whether OpenShift won't complain if you send a request with explicit `host: nil`, compared to "not mentioning" `host` at all. But if it works then good, it's definitely nicer then having if block.